### PR TITLE
Remove QUIC stuff, ddd docs

### PR DESCRIPTION
**Pull request by richsalz**
Remove all the one-liner/no-op QUIC functions after the big QUIC purge in PR #97.  Also remove much of the QUIC- specific documentation.

And also, because almost all of it was about QUIC, remove the demo-driven designs documentation.

Fixes: #108

This checklist is funny because I deleted stuff :)

## Checklist
Thank you for your contribution. Please answer the following:

- [x] I acknowledge that I am authorized to submit this code under
the terms of the [Apache License](https://www.apache.org/licenses/LICENSE-2.0)
- [ ] All new public APIs are documented
- [ ] All new public functions have tests
- [ ] All new public functions have fuzzing tests

### CODE REVIEW

File: doc/man3/SSL_get_value_uint.pod [hunk 1]
```diff
@@ -1,353 +0,0 @@
-=pod
-
-=head1 NAME
-
-SSL_get_value_uint, SSL_set_value_uint, SSL_get_generic_value_uint,
-SSL_set_generic_value_uint, SSL_get_feature_request_uint,
-SSL_set_feature_request_uint, SSL_get_feature_peer_request_uint,
-SSL_get_feature_negotiated_uint, SSL_get_quic_stream_bidi_local_avail,
```
Comment (botovq) [inline doc/man3/SSL_get_value_uint.pod:8]: I have no idea what this is supposed to be good for but you can never add too many knobs...

In any case, the first few of these seem generic and not necessarily quic related, so I'm unsure if these should be deleted.

**Review by botovq — COMMENTED** [review]
Mindboggling.

The PR as a whole looks good to me, but I'm not familiar with this new part of the OpenSSL tree.

File: doc/man3/SSL_get_value_uint.pod (hunk 1 above)
Comment (richsalz) [reply doc/man3/SSL_get_value_uint.pod:8]: The only thing they were ever used for is to query parameters on QUIC objects so I can't see a reason to keep them.

File: apps/s_client.c [hunk 2]
```diff
@@ -3002,19 +2985,15 @@ int s_client_main(int argc, char **argv)
              * they become readable/writeable and then SSL_handle_events() doing
              * the right thing.
              */
-            if ((!isquic && read_ssl)
-                    || (isquic && SSL_net_read_desired(con)))
+            if (!isquic && read_ssl)
```
Comment (wbl) [inline apps/s_client.c:2988]: !isquic is redundant here, as it will always be true.

File: doc/man3/SSL_get_rpoll_descriptor.pod [hunk 3]
```diff
@@ -2,8 +2,8 @@
 
 =head1 NAME
 
-SSL_get_rpoll_descriptor, SSL_get_wpoll_descriptor, SSL_net_read_desired,
-SSL_net_write_desired - obtain information which can be used to determine when
+SSL_get_rpoll_descriptor, SSL_get_wpoll_descriptor
```
Comment (wbl) [inline doc/man3/SSL_get_rpoll_descriptor.pod:5]: These functions also just make sense for quic, so this whole file can be deleted along with them.

File: doc/man3/SSL_handle_events.pod [hunk 4]
```diff
@@ -43,33 +37,6 @@ automatically by calls to other SSL functions such as L<SSL_read(3)> or
 L<SSL_write(3)>. Note that this is different to QUIC which also performs event
```
Comment (wbl) [inline doc/man3/SSL_handle_events.pod:37]: There is no longer a below to see

**Review by wbl — APPROVED** [review]
Mostly good, I think a few more functions can be removed.

File: doc/man3/SSL_handle_events.pod (hunk 4 above)
Comment (richsalz) [reply doc/man3/SSL_handle_events.pod:37]: will do in the next comment.

File: doc/man3/SSL_get_rpoll_descriptor.pod (hunk 3 above)
Comment (richsalz) [reply doc/man3/SSL_get_rpoll_descriptor.pod:5]: what about the `BIO_get[rw]poll_descriptor` functions?

File: doc/man3/SSL_handle_events.pod (hunk 4 above)
Comment (richsalz) [reply doc/man3/SSL_handle_events.pod:37]: I mean, I will fix this in the next PR

**Comment by richsalz** [comment]
Removing the the BIO polling stuff, too.  Not used.

**Pull request merged**